pkg/login/social/generic_oauth.go

```diff
@@ -412,28 +415,52 @@ func (s *SocialGenericOAuth) FetchPrivateEmail(client *http.Client) (string, err
 	return email, nil
 }
 
-func (s *SocialGenericOAuth) FetchTeamMemberships(client *http.Client) ([]int, bool) {
-	type Record struct {
-		Id int `json:"id"`
-	}
+func (s *SocialGenericOAuth) FetchTeamMemberships(client *http.Client) ([]string, bool) {
+	var response httpGetResponse
+	var err error
+	var ids []string
 
-	response, err := s.httpGet(client, fmt.Sprintf(s.apiUrl+"/teams"))
-	if err != nil {
-		s.log.Error("Error getting team memberships", "url", s.apiUrl+"/teams", "error", err)
-		return nil, false
-	}
+	if s.teamsUrl == "" {
+		s.log.Warn("[Deprecated] Switch to the use of the teams_url and team_id_attribute_path configuration options for filtering on team membership")
 
-	var records []Record
+		type Record struct {
+			Id int `json:"id"`
+		}
 
-	err = json.Unmarshal(response.Body, &records)
-	if err != nil {
-		s.log.Error("Error decoding team memberships response", "raw_json", string(response.Body), "error", err)
-		return nil, false
+		response, err = s.httpGet(client, fmt.Sprintf(s.apiUrl+"/teams"))
```

Comment:
This won't be possible (see: https://stackoverflow.com/questions/34668012/combine-url-paths-with-path-join/34668130)

Reply:
Fair enough.

I was trying to find a decent middleground because string concatenation on a URL path is shaky at best, but I didn't want you to have to do a bunch of refactoring to get this to be a url.URL.

Good callout, it's totally fine to leave it as-is and we'll have to refactor this to use a url.URL in the future.